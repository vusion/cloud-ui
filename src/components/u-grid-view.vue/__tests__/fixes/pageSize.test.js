import { mount } from '@vue/test-utils';

import PageSizeDemo from './__demos__/pageSize.vue';

const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

describe('u-grid-view.vue', () => {
    it('pageSizer选择器变化后，刷新数据', async () => {
        const wrapper = mount(PageSizeDemo);
        await sleep(16);

        const items = wrapper.findAll('div[index]');
        expect(items.length).toBe(10);

        // 页面选择器改为50后刷新数据
        wrapper.vm.$refs.gridview.$refs.pagination.onSelectPageSize({ value: 50 });
        const button = wrapper.find('#changelistbutton');
        await button.trigger('click');
        await sleep(16);
        const items1 = wrapper.findAll('div[index]');
        expect(items1.length).toBe(50);
        expect(wrapper.vm.$refs.gridview.currentData.length).toBe(50);
        expect(wrapper).toMatchSnapshot();

        // 将pageSize还原回20
        wrapper.vm.pageSize = 20;
        await sleep(16);
        const items2 = wrapper.findAll('div[index]');
        expect(items2.length).toBe(20);
        expect(wrapper.vm.$refs.gridview.currentData.length).toBe(20);

        expect(wrapper).toMatchSnapshot();
    });
});