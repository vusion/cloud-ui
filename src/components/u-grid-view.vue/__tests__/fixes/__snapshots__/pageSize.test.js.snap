// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`u-grid-view.vue pageSizer选择器变化后，刷新数据 1`] = `
<div class="root" direction="vertical" type vusion-slot-name="default">
  <div class="root" direction="horizontal" type vusion-slot-name="default">
    <a class="root" hovertype="underline" id="changelistbutton" tabindex="0" target="_self" vusion-slot-name-edit="text">
      变换list值
    </a>
  </div>
  <div border="true" class="root" tabindex="0">
    <div style="display: none;">
      <span vusion-slot-name="title">列表</span>
      <div>
      </div>
    </div>
    <div class="body">
      <div style="padding-top: 0px; padding-bottom: 0px;">
        <div class="root" index="0" style="width: 20%;" vusion-slot-name="item">Batch12
        </div>
        <div class="root" index="1" style="width: 20%;" vusion-slot-name="item">C12
        </div>
        <div class="root" index="2" style="width: 20%;" vusion-slot-name="item">C#12
        </div>
        <div class="root" index="3" style="width: 20%;" vusion-slot-name="item">C++12
        </div>
        <div class="root" index="4" style="width: 20%;" vusion-slot-name="item">CSS12
        </div>
        <div class="root" index="5" style="width: 20%;" vusion-slot-name="item">Clojure12
        </div>
        <div class="root" index="6" style="width: 20%;" vusion-slot-name="item">CoffeeScript12
        </div>
        <div class="root" index="7" style="width: 20%;" vusion-slot-name="item">Coq12
        </div>
        <div class="root" index="8" style="width: 20%;" vusion-slot-name="item">Diff12
        </div>
        <div class="root" index="9" style="width: 20%;" vusion-slot-name="item">Dockerfile12
        </div>
        <div class="root" index="10" style="width: 20%;" vusion-slot-name="item">F#12
        </div>
        <div class="root" index="11" style="width: 20%;" vusion-slot-name="item">Go12
        </div>
        <div class="root" index="12" style="width: 20%;" vusion-slot-name="item">Groovy12
        </div>
        <div class="root" index="13" style="width: 20%;" vusion-slot-name="item">HLSL12
        </div>
        <div class="root" index="14" style="width: 20%;" vusion-slot-name="item">HTML12
        </div>
        <div class="root" index="15" style="width: 20%;" vusion-slot-name="item">Handlebars12
        </div>
        <div class="root" index="16" style="width: 20%;" vusion-slot-name="item">Ignore12
        </div>
        <div class="root" index="17" style="width: 20%;" vusion-slot-name="item">Ini12
        </div>
        <div class="root" index="18" style="width: 20%;" vusion-slot-name="item">JSON12
        </div>
        <div class="root" index="19" style="width: 20%;" vusion-slot-name="item">Java12
        </div>
        <div class="root" index="20" style="width: 20%;" vusion-slot-name="item">JavaScript12
        </div>
        <div class="root" index="21" style="width: 20%;" vusion-slot-name="item">Jinja12
        </div>
        <div class="root" index="22" style="width: 20%;" vusion-slot-name="item">Jupyter12
        </div>
        <div class="root" index="23" style="width: 20%;" vusion-slot-name="item">Less12
        </div>
        <div class="root" index="24" style="width: 20%;" vusion-slot-name="item">Log12
        </div>
        <div class="root" index="25" style="width: 20%;" vusion-slot-name="item">Lua12
        </div>
        <div class="root" index="26" style="width: 20%;" vusion-slot-name="item">Makefile12
        </div>
        <div class="root" index="27" style="width: 20%;" vusion-slot-name="item">Markdown12
        </div>
        <div class="root" index="28" style="width: 20%;" vusion-slot-name="item">Objective-C12
        </div>
        <div class="root" index="29" style="width: 20%;" vusion-slot-name="item">Objective-C++12
        </div>
        <div class="root" index="30" style="width: 20%;" vusion-slot-name="item">PHP12
        </div>
        <div class="root" index="31" style="width: 20%;" vusion-slot-name="item">Perl12
        </div>
        <div class="root" index="32" style="width: 20%;" vusion-slot-name="item">PowerShell12
        </div>
        <div class="root" index="33" style="width: 20%;" vusion-slot-name="item">Properties12
        </div>
        <div class="root" index="34" style="width: 20%;" vusion-slot-name="item">Pug12
        </div>
        <div class="root" index="35" style="width: 20%;" vusion-slot-name="item">Python12
        </div>
        <div class="root" index="36" style="width: 20%;" vusion-slot-name="item">R12
        </div>
        <div class="root" index="37" style="width: 20%;" vusion-slot-name="item">Razor12
        </div>
        <div class="root" index="38" style="width: 20%;" vusion-slot-name="item">Ruby12
        </div>
        <div class="root" index="39" style="width: 20%;" vusion-slot-name="item">Rust12
        </div>
        <div class="root" index="40" style="width: 20%;" vusion-slot-name="item">SCSS12
        </div>
        <div class="root" index="41" style="width: 20%;" vusion-slot-name="item">SQL12
        </div>
        <div class="root" index="42" style="width: 20%;" vusion-slot-name="item">SVG12
        </div>
        <div class="root" index="43" style="width: 20%;" vusion-slot-name="item">Shaderlab12
        </div>
        <div class="root" index="44" style="width: 20%;" vusion-slot-name="item">Shell Script12
        </div>
        <div class="root" index="45" style="width: 20%;" vusion-slot-name="item">Swift12
        </div>
        <div class="root" index="46" style="width: 20%;" vusion-slot-name="item">TypeScript12
        </div>
        <div class="root" index="47" style="width: 20%;" vusion-slot-name="item">Visual Basic12
        </div>
        <div class="root" index="48" style="width: 20%;" vusion-slot-name="item">Vue12
        </div>
        <div class="root" index="49" style="width: 20%;" vusion-slot-name="item">XML12
        </div>
      </div>
    </div>
    <div class="foot">
      <nav class="root pagination" size="normal">
        <div class="root sizer" tabindex="0">
          <span>b</span>
          <span style="display: none;">请选择</span>
          <div class style="direction: ltr;" tags-overflow="collapse">
            50条/页
          </div>
          <transition-stub name="fade" style="width: 100%;">
            <div class="root" style="display: none;">
              <div>
                <div class="root" style="direction: ltr;" vusion-slot-name="default">
                  10条/页
                </div>
                <div class="root" style="direction: ltr;" vusion-slot-name="default">
                  20条/页
                </div>
                <div class="root" selected="selected" style="direction: ltr;" vusion-slot-name="default">
                  50条/页
                </div>
              </div>
            </div>
          </transition-stub>
        </div>
        <a class="item-wrap" disabled="disabled">
          <i class="item" role="prev"></i>
        </a>
        <a class="item" selected="selected">1</a>
        <a class="item">2</a>
        <a class="item-wrap">
          <i class="item" role="next"></i>
        </a>
      </nav>
    </div>
  </div>
</div>
`;

exports[`u-grid-view.vue pageSizer选择器变化后，刷新数据 2`] = `
<div class="root" direction="vertical" type vusion-slot-name="default">
  <div class="root" direction="horizontal" type vusion-slot-name="default">
    <a class="root" hovertype="underline" id="changelistbutton" tabindex="0" target="_self" vusion-slot-name-edit="text">
      变换list值
    </a>
  </div>
  <div border="true" class="root" tabindex="0">
    <div style="display: none;">
      <span vusion-slot-name="title">列表</span>
      <div>
      </div>
    </div>
    <div class="body">
      <div style="padding-top: 0px; padding-bottom: 0px;">
        <div class="root" index="0" style="width: 20%;" vusion-slot-name="item">Batch12
        </div>
        <div class="root" index="1" style="width: 20%;" vusion-slot-name="item">C12
        </div>
        <div class="root" index="2" style="width: 20%;" vusion-slot-name="item">C#12
        </div>
        <div class="root" index="3" style="width: 20%;" vusion-slot-name="item">C++12
        </div>
        <div class="root" index="4" style="width: 20%;" vusion-slot-name="item">CSS12
        </div>
        <div class="root" index="5" style="width: 20%;" vusion-slot-name="item">Clojure12
        </div>
        <div class="root" index="6" style="width: 20%;" vusion-slot-name="item">CoffeeScript12
        </div>
        <div class="root" index="7" style="width: 20%;" vusion-slot-name="item">Coq12
        </div>
        <div class="root" index="8" style="width: 20%;" vusion-slot-name="item">Diff12
        </div>
        <div class="root" index="9" style="width: 20%;" vusion-slot-name="item">Dockerfile12
        </div>
        <div class="root" index="10" style="width: 20%;" vusion-slot-name="item">F#12
        </div>
        <div class="root" index="11" style="width: 20%;" vusion-slot-name="item">Go12
        </div>
        <div class="root" index="12" style="width: 20%;" vusion-slot-name="item">Groovy12
        </div>
        <div class="root" index="13" style="width: 20%;" vusion-slot-name="item">HLSL12
        </div>
        <div class="root" index="14" style="width: 20%;" vusion-slot-name="item">HTML12
        </div>
        <div class="root" index="15" style="width: 20%;" vusion-slot-name="item">Handlebars12
        </div>
        <div class="root" index="16" style="width: 20%;" vusion-slot-name="item">Ignore12
        </div>
        <div class="root" index="17" style="width: 20%;" vusion-slot-name="item">Ini12
        </div>
        <div class="root" index="18" style="width: 20%;" vusion-slot-name="item">JSON12
        </div>
        <div class="root" index="19" style="width: 20%;" vusion-slot-name="item">Java12
        </div>
      </div>
    </div>
    <div class="foot">
      <nav class="root pagination" size="normal">
        <div class="root sizer" tabindex="0">
          <span>b</span>
          <span style="display: none;">请选择</span>
          <div class style="direction: ltr;" tags-overflow="collapse">
            20条/页
          </div>
          <transition-stub name="fade" style="width: 100%;">
            <div class="root" style="display: none;">
              <div>
                <div class="root" style="direction: ltr;" vusion-slot-name="default">
                  10条/页
                </div>
                <div class="root" selected="selected" style="direction: ltr;" vusion-slot-name="default">
                  20条/页
                </div>
                <div class="root" style="direction: ltr;" vusion-slot-name="default">
                  50条/页
                </div>
              </div>
            </div>
          </transition-stub>
        </div>
        <a class="item-wrap" disabled="disabled">
          <i class="item" role="prev"></i>
        </a>
        <a class="item" selected="selected">1</a>
        <a class="item">2</a>
        <a class="item">3</a>
        <a class="item-wrap">
          <i class="item" role="next"></i>
        </a>
      </nav>
    </div>
  </div>
</div>
`;
